### PR TITLE
Fix parsing of aws-cli expiration timestamp

### PR DESCRIPTION
**Pull request by skodde**
aws-cli seems to return the timestamp in ISO-8601 format, let's use
dateutil.parser to convert it in a datetime.datetime object.

##### ENVIRONMENTS AFFECTED
None

### CODE REVIEW

File: src/commcare_cloud/commands/terraform/aws.py [hunk 1]
```diff
@@ -583,8 +584,8 @@ def _has_valid_session_credentials_for_sso():
             continue
 
         if 'startUrl' in contents and 'expiresAt' in contents:
-            expiration = datetime.strptime(contents['expiresAt'], "%Y-%m-%dT%H:%M:%SUTC")
-            return datetime.utcnow() < expiration
```
Comment (calellowitz) [inline src/commcare_cloud/commands/terraform/aws.py:587]: why remove this comparison? I think it checks whether the credentials are still valid so we do not need to auth again.

File: src/commcare_cloud/commands/terraform/aws.py [hunk 2]
```diff
@@ -598,11 +599,11 @@ def _has_valid_v1_session_credentials(aws_profile):
     if aws_profile not in config.sections():
         return False
     try:
-        expiration = datetime.strptime(config.get(aws_profile, 'expiration'), "%Y-%m-%dT%H:%M:%SZ")
+        expiration = parser.isoparse(config.get(aws_profile, 'expiration'))
     except configparser.NoOptionError:
         return False
 
-    return datetime.utcnow() < expiration
```
Comment (calellowitz) [inline src/commcare_cloud/commands/terraform/aws.py:605]: same question about the comparison

**Review by calellowitz — COMMENTED** [review]
Left one question but otherwise looks great. I hadn't seen `isoparse` before, just `parse` but if that only checks ISO compatible formats, its a nice safety feature

File: src/commcare_cloud/commands/terraform/aws.py [hunk 3]
```diff
@@ -583,8 +584,8 @@ def _has_valid_session_credentials_for_sso():
             continue
 
         if 'startUrl' in contents and 'expiresAt' in contents:
-            expiration = datetime.strptime(contents['expiresAt'], "%Y-%m-%dT%H:%M:%SUTC")
-            return datetime.utcnow() < expiration
+            expiration = parser.isoparse(contents['expiresAt'])
+            return expiration
```
Comment (millerdev) [inline src/commcare_cloud/commands/terraform/aws.py:588]: I think this should return a boolean, not a date. Did you remove the comparison by mistake?

File: src/commcare_cloud/commands/terraform/aws.py [hunk 4]
```diff
@@ -598,11 +599,11 @@ def _has_valid_v1_session_credentials(aws_profile):
     if aws_profile not in config.sections():
         return False
     try:
-        expiration = datetime.strptime(config.get(aws_profile, 'expiration'), "%Y-%m-%dT%H:%M:%SZ")
+        expiration = parser.isoparse(config.get(aws_profile, 'expiration'))
     except configparser.NoOptionError:
         return False
 
-    return datetime.utcnow() < expiration
+    return expiration
```
Comment (millerdev) [inline src/commcare_cloud/commands/terraform/aws.py:606]: Same as above comment.

**Comment by skodde** [comment]
Fixed the comparison in the second commit

**Review by calellowitz — APPROVED** [review]

File: src/commcare_cloud/commands/terraform/aws.py [hunk 5]
```diff
@@ -439,7 +438,7 @@ def _sync_sso_to_v1_credentials(aws_session_profile):
         aws_access_key_id=credentials['AccessKeyId'],
         aws_secret_access_key=credentials['SecretAccessKey'],
         aws_session_token=credentials['SessionToken'],
-        expiration = parser.isoparse(credentials['Expiration']),
+        expiration=parser.isoparse(credentials['Expiration']),
```
Comment (calellowitz) [inline src/commcare_cloud/commands/terraform/aws.py:441]: I think this might have to be `parser.parse` to handle both formats

**Review by millerdev — APPROVED** [review]

**Review by calellowitz — APPROVED** [review]

**Review by dannyroberts — APPROVED** [review]

File: src/commcare_cloud/commands/terraform/aws.py (hunk 5 above)
Comment (skodde) [reply src/commcare_cloud/commands/terraform/aws.py]: Fixed in a previous commit.

File: src/commcare_cloud/commands/terraform/aws.py (hunk 3 above)
Comment (skodde) [reply src/commcare_cloud/commands/terraform/aws.py]: See https://github.com/dimagi/commcare-cloud/pull/4455#issuecomment-759717915

File: src/commcare_cloud/commands/terraform/aws.py (hunk 4 above)
Comment (skodde) [reply src/commcare_cloud/commands/terraform/aws.py]: See https://github.com/dimagi/commcare-cloud/pull/4455#issuecomment-759717915

File: src/commcare_cloud/commands/terraform/aws.py (hunk 1 above)
Comment (skodde) [reply src/commcare_cloud/commands/terraform/aws.py:587]: See https://github.com/dimagi/commcare-cloud/pull/4455#issuecomment-759717915

File: src/commcare_cloud/commands/terraform/aws.py (hunk 2 above)
Comment (skodde) [reply src/commcare_cloud/commands/terraform/aws.py:605]: See https://github.com/dimagi/commcare-cloud/pull/4455#issuecomment-759717915

**Review by millerdev — APPROVED** [review]

**Pull request merged**